site/gdocs/components/HomepageIntro.tsx

```diff
@@ -116,6 +118,87 @@ function FeaturedWorkTile({
     )
 }
 
+function useIsOverflowing<T extends HTMLElement = HTMLElement>(
+    ref: RefObject<T>
+): boolean {
+    const [isOverflowing, setIsOverflowing] = useState(false)
+
+    useResizeObserver({
+        ref,
+        onResize: () => {
+            const element = ref.current
+            if (!element) return
+
+            const overflowing =
+                element.scrollWidth > element.clientWidth ||
+                element.scrollHeight > element.clientHeight
+
+            setIsOverflowing(overflowing)
+        },
+    })
+
+    return isOverflowing
+}
+
+function HomepageAnnouncement(props: {
+    announcement: OwidGdocMinimalPostInterface
+    index: number
+}) {
+    const { announcement, index } = props
+    const { linkedChart } = useLinkedChart(announcement.cta?.url || "")
+    const { linkedDocument } = useLinkedDocument(announcement.cta?.url || "")
+    // If there's not enough space to show all of the third announcement, we link to the latest page, no matter what
+    // If there is enough space, it behaves as normal:
+    // - cta: link directly to the resource
+    // - no cta: link to the latest page
+    const announcementRef = useRef<HTMLLIElement>(null)
+    const isOverflowing = useIsOverflowing(
+        announcementRef as RefObject<HTMLElement>
+    )
+    const latestPageLink = `/latest#${announcement.slug}`
+    const href = isOverflowing
+        ? latestPageLink
+        : linkedChart?.resolvedUrl ||
+          linkedDocument?.url ||
+          announcement.cta?.url ||
+          latestPageLink
+
+    return (
+        <li
+            className={cx("homepage-intro__announcement", {
+                "homepage-intro__announcement--is-overflowing": isOverflowing,
+            })}
+            key={announcement.id}
```

Comment:
There should be no key here, and instead it should be `announcement.id` in the `HomepageAnnouncements` where we use this component.